app/controllers/comments_controller.rb

```diff
@@ -1,13 +1,42 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
+
 def create
   @post = Post.find(params[:post_id])
```

Comment:
This line is causing problems in your tests.  Right now you have:

```
   @post = Post.find(params[:post_id])
```

Which raises an error if something isn't found.  Instead we could do something like:

```
@post = Post.find_by_id(params[:post_id])
render_not_found if @post.blank?
```

To fix it.